DNN Platform/Library/Entities/Users/UserEventHandlers.cs

```diff
@@ -51,6 +51,7 @@ public void UserDeleted(object sender, UserEventArgs args)
 
         public void UserApproved(object sender, UserEventArgs args)
         {
+            if (!args.SendNotification) return;
             Mail.SendMail(args.User, MessageType.UserRegistrationPublic, PortalSettings.Current);
             DeleteAllNotifications(args.User.UserID);
```

Comment:
Does `DeleteAllNotifications` not need to happen if we aren't sending a new notification?